### ISSUE TITLE
Additional/alternative solution to `ShortcodeTest`?

### ISSUE THREAD

**Issue by mpdude** [webfactory/WebfactoryShortcodeBundle#13]
The `ShortcodeTest` base class is a full-fledged `WebTestCase`. It assumes that you have set up the [shortcode guide](https://github.com/webfactory/WebfactoryShortcodeBundle#activating-the-shortcode-guide), since it uses the rendering feature provided by that to display the shortcode under test in isolation.

That may be a bit complicated to use, for example when rendering the shortcode requires staging some database content. Imagine a shortcode that fetches some database rows and renders a table with it.

In such a case, you might not want to put different test data into the database, then run a `ShortcodeTest` with your shortcode and inspect the resulting HTML for correctness. It may be easier to directly test the controller generating the shortcode substitution HTML, or even the sub-modules or components that controller uses.

That leaves you with two types of tests:

1. For a given shortcode (name + optionally parameters), verify that the current configuration (DIC, tagged services etc.) would actually call the right controller with the right parameters when filtering content through `|shortcodes`; and that the HTML (or whatever response body returned) is actually embedded instead of the original shortcode
2. Test that the controllers generating shortcode substitution content work as expected for various datasets/inputs.

Regarding 1.), you could even argue that the substitution part is not worth testing if you rely on this bundle here doing its job properly. What you would like to assert as a user of this bundle is that you've done your configuration and service tagging correctly and that a shortcode you expect to work is registered and being processed.

(Note that even this test does not guarantee you use the `|shortcodes` filter where appropriate; this seems to be completely outside the scope of this bundle?)

**Bottom line:** Do we want to provide an additional/alternative test base class that supports testing the configuration works as expected, without actually running the controllers generating the shortcode substitution content, and without requiring the setup of the *shortcode guide* feature?
